safe/utilities/analysis.py

```diff
@@ -125,12 +125,16 @@ def user_extent(self):
     @user_extent.setter
     def user_extent(self, user_extent):
         # We transfer the extent to the IF as a list.
-        extent = [
-            user_extent.xMinimum(),
-            user_extent.yMinimum(),
-            user_extent.xMaximum(),
-            user_extent.yMaximum()]
-        self.impact_function.requested_extent = extent
+        if isinstance(user_extent, QgsRectangle):
+            extent = [
+                user_extent.xMinimum(),
+                user_extent.yMinimum(),
+                user_extent.xMaximum(),
+                user_extent.yMaximum()]
+            self.impact_function.requested_extent = extent
+        else:
```

Comment:
It should never happen in the current state of InaSAFE.
The check is done before everytime. I put this exception **in case of** this function doesn't get a `QgsRectangle` or `None`. As I said this `analysis.py` will be removed soon.